vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -1740,3 +1741,624 @@ def execute_model(
+        req_id_to_index = self.get_req_id_to_index(is_prefill)
+
+        prompt_logprobs_dicts = self._get_prompt_logprobs_dict(
+            logits=logits, model_inputs=model_input)
+
+        # If the prompt is being prefilled we don't have to sample
+        # and generate a new token.
+        if is_prefill and self.check_incomplete_prefill(scheduler_output):
+            # Only return outputs from the driver worker
+            if not self.is_driver_worker:
+                return self.get_empty_output()
+
+            return CBSpyreModelRunnerOutput(
+                req_ids=list(req_id_to_index.keys()),
+                req_id_to_index=req_id_to_index,
+                sampled_token_ids=[],
+                logprobs=None,
+                # TODO: probably it does not makes sense
+                prompt_logprobs_dict=prompt_logprobs_dicts,
+                pooler_output=[],
+                tkv=self.tkv,
+                n_free_blocks=self.get_n_free_blocks())
+
+        # Sample the next token.
+        output: SamplerOutput = self.model.sample(
+            logits=logits,
+            sampling_metadata=self.get_sampling_metadata(is_prefill),
+        )
+        t1 = time.time() - t0
+        logger.debug("t_token: %.2fms", (t1 * 1000))
+
+        # Add the sampled token(s) to the request cache
+        req_ids = ([r.req_id for r in scheduler_output.scheduled_new_reqs]
+                   if len(scheduler_output.scheduled_new_reqs) > 0 \
+                    else self.input_batch.sorted_requests_ids)
+
+        # Get the right batch, if this is the last chunk to conclude the
+        # prefill, we'll generate a token and we should get from the prefill
+        # batch because input_batch may have other request that are were
+        # not processed at this step.
+        batch = self.prefill_batch if is_prefill \
+            else self.input_batch
+
+        # Add the sampled token(s) to the request cache
+        req_ids = ([r.req_id for r in scheduler_output.scheduled_new_reqs]
+                if len(scheduler_output.scheduled_new_reqs) > 0 \
+                else batch.sorted_requests_ids)
+        sampled_ids = output.sampled_token_ids.tolist()
+
+        for i, req_id in enumerate(req_ids):
+            req_state = self.requests[req_id]
+            req_state.output_token_ids.extend(sampled_ids[i])
+
+        # Only return outputs from the driver worker
+        if not self.is_driver_worker:
+            return self.get_empty_output()
+
+        model_output = CBSpyreModelRunnerOutput(
+            req_ids=list(req_id_to_index.keys()),
+            req_id_to_index=req_id_to_index,
+            sampled_token_ids=output.sampled_token_ids.tolist(),
+            logprobs=(output.logprobs_tensors.tolists()
+                      if output.logprobs_tensors else None),
+            prompt_logprobs_dict=prompt_logprobs_dicts,
+            pooler_output=[],
+            tkv=self.tkv,
+            n_free_blocks=self.get_n_free_blocks())
```

Comment:
Would it be possible still in this PR to send the correct number of `num_computed_tokens` for the prefilling request back to the scheduler? From the discussions it seems like the best way would be through `CBSpyreModelRunnerOutput`. See [my comment in the scheduler changes](https://github.com/vllm-project/vllm-spyre/blob/085068eb2907893210c49faef541ce9620c154dd/vllm_spyre/v1/core/scheduler.py#L436-L439).